misc/version_updater.py

```diff
@@ -0,0 +1,325 @@
+            ),
+            Replace(r"RUST_TOOLCHAIN=\$\{1:-.*\}", "RUST_TOOLCHAIN=${{1:-{version}}}"),
+        ]
+    },
+    ROOT_DIR
+    / "misc/version_updater.py": {
+        Tool.Rust: [Replace(r'Tool.Rust: "[0-9.]+"', 'Tool.Rust: "{version}"')],
+        Tool.Python: [Replace(r'Tool.Python: "[0-9.]+"', 'Tool.Python: "{version}"')],
+        Tool.Poetry: [Replace(r'Tool.Poetry: "[0-9.]+"', 'Tool.Poetry: "{version}"')],
+        Tool.Node: [Replace(r'Tool.Node: "[0-9.]+"', 'Tool.Node: "{version}"')],
+        Tool.WasmPack: [Replace(r'Tool.WasmPack: "[0-9.]+"', 'Tool.WasmPack: "{version}"')],
+    },
+    ROOT_DIR
+    / "packaging/server/res/build-backend.sh": {
+        Tool.Python: [Replace(r"python\d.\d+", hide_patch_version("python{version}"))]
+    },
+    ROOT_DIR
+    / "packaging/server/server.dockerfile": {
+        Tool.Python: [Replace(r"python:\d.\d+", hide_patch_version("python:{version}"))]
+    },
+    ROOT_DIR
+    / "packaging/snap/bin/parsec": {
+        Tool.Python: [Replace(r"python\d.\d+", hide_patch_version("python{version}"))]
+    },
+    ROOT_DIR
+    / "packaging/snap/bin/parsec-cli": {
+        Tool.Python: [Replace(r"python\d.\d+", hide_patch_version("python{version}"))]
+    },
+    ROOT_DIR
+    / "packaging/snap/snap/snapcraft.yaml": {
+        Tool.Python: [Replace(r"python\d.\d+", hide_patch_version("python{version}"))]
+    },
+    ROOT_DIR
+    / "packaging/testbed-server/build-testbed.sh": {
+        Tool.Python: [Replace(r"python\d.\d+", hide_patch_version("python{version}"))]
+    },
+    ROOT_DIR
+    / "packaging/testbed-server/testbed-server.dockerfile": {
+        Tool.Python: [Replace(r"python:\d.\d+", hide_patch_version("python:{version}"))]
+    },
+    ROOT_DIR
+    / "mypy.ini": {
+        Tool.Python: [
+            Replace(r"python_version = \d.\d+", hide_patch_version("python_version = {version}"))
+        ]
+    },
+    ROOT_DIR
+    / ".pre-commit-config.yaml": {
+        Tool.Rust: [Replace(r"rust: [0-9.]+", "rust: {version}")],
+        Tool.Node: [Replace(r"node: [0-9.]+", "node: {version}")],
+    },
+    ROOT_DIR
+    / "pyproject.toml": {
+        Tool.Python: [
+            Replace(
+                r'"Programming Language :: Python :: .*"',
+                hide_patch_version('"Programming Language :: Python :: {version}"'),
+            ),
+            Replace(r'python = "~.*"', 'python = "~{version}"'),
+            Replace(
+                r'build = "cp\d+-{manylinux,macos,win}\*"',
+                hide_patch_version('build = "cp{version}-{{manylinux,macos,win}}*"', separator=""),
+            ),
+            Replace(r"py\d+", hide_patch_version("py{version}", separator="")),
+        ]
+    },
+    ROOT_DIR
+    / "rust-toolchain.toml": {Tool.Rust: [Replace(r'channel = ".*"', 'channel = "{version}"')]},
+}
+
+
+def check_correct_tool_version(filename: Path, raw_regexes: RawRegexes, version: str) -> List[str]:
```

Comment:
What about:
```suggestion
def check_tool_version(filename: Path, raw_regexes: RawRegexes, version: str) -> List[str]:
```
to be more consistent with `update_tool_version` name?

Reply:
done